lib/sass_spec/engine_adapter.rb

```diff
@@ -54,8 +54,11 @@ def version
 
 
   def compile(sass_filename, precision)
-    cmd = "#{@command} --precision #{precision} -t expanded"
-    result = capture3_with_timeout("#{cmd} #{sass_filename}", :binmode => true, :timeout => @timeout)
+    cmd = "#{File.absolute_path(@command)} --precision #{precision} -t expanded"
```

Comment:
This won't work if `File.absolute_path(@command)` contains spaces.

While you could manually escape the path like this:

```ruby
"#{Shellwords.escape File.absolute_path(@command)}"
```

Consider simply passing the arguments individually:

```ruby
capture3_with_timeout(
  File.absolute_path(@command), '--precision', precision, '-t', expanded,
  binmode: true, timeout: timeout
)
```

(see https://ruby-doc.org/stdlib-1.9.3/libdoc/open3/rdoc/Open3.html#method-c-popen3)

Reply:
Also, not sure why we call `File.absolute_path` here, capture3 is already invoked in `$PWD`.

Reply:
> This won't work if `File.absolute_path(@command)` contains spaces.
> 
> While you could manually escape the path like this:
> 
> ```ruby
> "#{Shellwords.escape File.absolute_path(@command)}..."
> ```

Done.

> Consider simply passing the arguments individually:
> 
> ```ruby
> capture3_with_timeout(
>   File.absolute_path(@command), '--precision', precision, '-t', expanded,
>   binmode: true, timeout: timeout
> )
> ```
> 
> (see https://ruby-doc.org/stdlib-1.9.3/libdoc/open3/rdoc/Open3.html#method-c-popen3)

This would require rewriting `capture3_with_timeout` to avoid using `spawn()`, which is probably a good idea, but not something I'd like to lump into this PR.

> I'm also not sure why we call `File.absolute_path` here, capture3 is already invoked in `$PWD`.

We change the PWD with `Dir.chdir`, so we need to make sure the command path remains valid in the new directory.

Reply:
`spawn` also support this!

See https://ruby-doc.org/core-2.2.0/Process.html#method-c-spawn:

>   cmdname, arg1, ...          : command name and one or more arguments (This form does not use the shell. See below for caveats.)

Reply:
Oh neat! I must have misread the API docs. Done.